### PR TITLE
USER_ID() should be case-insensitive.

### PR DESCRIPTION
**Pull request by timchang514**
USER_ID() needs to handle case-insensitive input by lowercasing them. A similar
change was already applied to db_id().

Task: BABEL-3223
Signed-off-by: Tim Chang <changti@amazon.com>

### Description

[Describe what this change achieves]
 
### Issues Resolved

[List any issues this PR will resolve]

### Check List
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is under the terms of the Apache 2.0 and PostgreSQL licenses, and grant any person obtaining a copy of the contribution permission to relicense all or a portion of my contribution to the PostgreSQL License solely to contribute all or a portion of my contribution to the PostgreSQL open source project.

For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/babelfish-for-postgresql/babelfish_extensions/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by xhfanhe — DISMISSED** [review]

File: contrib/babelfishpg_tsql/src/rolecmds.c [hunk 1]
```diff
@@ -684,6 +685,10 @@ user_id(PG_FUNCTION_ARGS)
 	if (!user_name)
 		PG_RETURN_NULL();
 
+	if (pltsql_case_insensitive_identifiers)
+        // Lowercase the entry, if needed
+        for (char *p = user_name ; *p; ++p) *p = tolower(*p);
```
Comment (Deepesh125) [inline contrib/babelfishpg_tsql/src/rolecmds.c]: Please fix the indentation.

Why can't we use downcase_identifier?

File: test/JDBC/input/BABEL-LOGIN-USER-EXT.mix [hunk 2]
```diff
@@ -327,6 +327,9 @@ GO
 SELECT name FROM sys.server_principals WHERE principal_id = suser_id('login1    ');
```
Comment (Deepesh125) [inline test/JDBC/input/BABEL-LOGIN-USER-EXT.mix:387]: Add upgrade tests also to verify before and after behaviour verification.

File: contrib/babelfishpg_tsql/src/rolecmds.c (hunk 1 above)
Comment (timchang514) [reply contrib/babelfishpg_tsql/src/rolecmds.c]: I had an offline conversation with Sangil about this with the previous fix https://code.amazon.com/reviews/CR-69414431/revisions/2#/comments - downcase_identifier caused segfaults in some scenarios when being used here.

File: test/JDBC/input/BABEL-LOGIN-USER-EXT.mix (hunk 2 above)
Comment (timchang514) [reply test/JDBC/input/BABEL-LOGIN-USER-EXT.mix:387]: Done

**Review by macwoody — APPROVED** [review]

**Review by priyansx — APPROVED** [review]
LGTM

**Review by macwoody — APPROVED** [review]

**Pull request merged**